File: hospitalManagement/NonEmergencyTransportation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace hospitalManagement
{
    internal abstract class NonEmergencyTransportation : MovementEquipment
    {

        //Field
        private string clientRequirement;
        private string purpose;
        // Infor field
        //  Management field
        // Dynamic field
        // Properties
        public string ClientRequiremnt
        {
            get => clientRequirement; set => clientRequirement = value;
        }
        public string Purpose
        {
            get => purpose; set => purpose = value;
        }

        // Constructors
        public NonEmergencyTransportation(
           string id
           , string name
           , string description
           , float price
           , bool state
           , string origin
           , int quantity
           , int numberOfSeat
           , string destination
            , string clientRequiremnt
            , string purpose
           ) : base(id, name, description, price, state, origin, quantity, numberOfSeat, destination)
        {
            ClientRequiremnt = clientRequiremnt;
            Purpose = purpose;
        }

        protected NonEmergencyTransportation()
        {
        }

        // Destructors
        ~NonEmergencyTransportation() { }
        // Methods
        // in, output
        public override void Input()
        {
            base.Input();
            Console.Write("Requirement: ");
            ClientRequiremnt = Console.ReadLine();
            Console.Write("Purpose: ");
            purpose = Console.ReadLine();
        }
        public override void Output()
        {
            base.Output();
            Console.WriteLine($"Requirement: {clientRequirement}");
            Console.WriteLine($"Purpose: {purpose}");
        }
        // General method
        // Other method
        // Overriding
        public override string ToString()
        {
            return base.ToString()
                + $"Requirement: {clientRequirement}"
                + $"Client's purpose: {purpose}";
        }
    }
}

File: src/hospitalManagement/NonEmergencyTransportation.java
package hospitalManagement;

import java.util.Scanner;

public abstract class NonEmergencyTransportation extends MovementEquipment {
	//Field
    private String clientRequirement;
    private String purpose;
    // Information field
    //  Management field
    // Dynamic field
    // Properties
    

    // Constructors
    public String getClientRequirement() {
		return clientRequirement;
	}

	public void setClientRequirement(String clientRequirement) {
		this.clientRequirement = clientRequirement;
	}

	public String getPurpose() {
		return purpose;
	}

	public void setPurpose(String purpose) {
		this.purpose = purpose;
	}

	public NonEmergencyTransportation(
       String id
       , String name
       , String description
       , float price
       , boolean state
       , String origin
       , int quantity
       , int numberOfSeat
       , String destination
        , String clientRequirement
        , String purpose
       )
    {
		super(id, name, description, price, state, origin, quantity, numberOfSeat, destination);
        this.clientRequirement = clientRequirement;
        this.purpose = purpose;
    }

    public NonEmergencyTransportation()
    {
    }

    // Destructors

    // Methods
    // in, output
    public  void Input()
    {
        super.Input();
        Scanner ip = new Scanner(System.in);
        System.out.print("Requirement: ");
        clientRequirement = ip.nextLine();
        System.out.print("Purpose: ");
        purpose = ip.nextLine();
        ip.close();
    }
    public  void Output()
    {
        super.Output();
        System.out.format("Requirement: %s", clientRequirement);
        System.out.format("Purpose: %s", purpose);
    }
    // General method
    // Other method
    // Overriding
    public  String ToString()
    {
        return super.ToString()
            + "Requirement: " + clientRequirement
            + "Client's purpose: " + purpose;
    }
}
